### PR TITLE
fix: Preserve return_to param between flows

### PR DESCRIPTION
**Pull request by mmeller-wikia**
Submitted on behalf of a third-party: [fandom.com](http://fandom.com/)

This changes fixes redirects for OIDC register/login flows.
#### Registration
* Create new account using OIDC provider
* Logout
* Go to some page (allowed for redirect)
* Try to register new account with the same OIDC → you should be logged in and redirected to previous page
#### Login
* Go to some page (allowed for redirect)
* Try to signin with OIDC (not linked to any user)
* Register new account -> you should be logged in and redirected to previous page

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I am following the [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security. vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

**Comment by mmeller-wikia** [comment]
Hi @aeneasr 
Can I ask for review?
I think that failing checks are not related to my changes ;)

File: x/http_secure_redirect.go [hunk 1]
```diff
@@ -73,6 +73,26 @@ func SecureRedirectToIsAllowedHost(returnTo *url.URL, allowed url.URL) bool {
 	return strings.EqualFold(allowed.Host, returnTo.Host)
 }
 
+func TakeOverReturnToParameter(from string, to string) (string, error) {
```
Comment (aeneasr) [inline x/http_secure_redirect.go:76]: Can you please add a test for this?

**Review by aeneasr — CHANGES_REQUESTED** [review]
Thank you, great job! Can you please provide a failing test case for the bug you observed to prove that this change is effective?

**Review by aeneasr — APPROVED** [review]

**Comment by aeneasr** [comment]
Awesome, thank you! 🎉 Your contribution makes Ory better :)

**Comment by vinckr** [comment]
Hello @mmeller-wikia 
Congrats on merging your first PR in Ory 🎉 !
Your contribution will soon be helping secure millions of identities around the globe 🌏. 
As a small token of appreciation we send all our first time contributors a gift package to welcome them to the community. 
[Please drop me an email and I will forward you the form to claim your Ory swag!](mailto:vincent@ory.sh)

**Pull request merged**